templates/core/Resources/_ti/extension.loader.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2018 by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ * 
+ * Description:
+ * Loads all of Titanium's core JavaScript files under the "extensions" directory.
+ * Does not load JS files recursively. Does a simple flat directory search instead.
+ * 
+ * This loader is expected to be ran before executing the "app.js" file. Intended to
+ * share Titanium's core JavaScript code/extensions on all platforms.
+ */
+
+'use strict';
+
+var JS_EXTENSION = '.js',
+	DIRECTORY_PATH = '_ti/extensions/';
+
+exports.load = function () {
```

Comment:
There's a lot of code in this function that can be simplified.

I'm not sure why you are checking if `fileNames` and `fileName` is truthy. Why would it not be? Isn't that a bug if `getDirectoryListing()` returns something falsey?

Use a for-of loop. It's faster and cuts down on all the cruft.

Use a regex to check if the file is a `.js` file. It's way cleaner and faster than doing all that string manipulation.
```js
const DIRECTORY_PATH = '_ti/extensions/';

exports.load = function load() {
	const dir = Ti.Filesystem.getFile(Ti.Filesystem.getResourcesDirectory(), DIRECTORY_PATH);
	const jsRegExp = /^(.*)\.js$/;

	for (const filename of directory.getDirectoryListing().sort()) {
		const m = filename.match(jsRegExp);
		if (m) {
			require(DIRECTORY_PATH + m[1]);
		}
	}
};
```

Reply:
We have a parity issue with `getDirectoryListing()`. If the directory is empty, iOS returns `null` and Android returns an empty array. Personally, I prefer the empty array since the app developer may forget to do a `null` check. CC: @hansemannn 

I benchmarked the regex approach. The performance was about the same, but the regex approach is simpler so I'll roll with it. Thanks man.

Reply:
Sounds like an easy parity fix for iOS. I'll take a quick look today.

Reply:
I found Ti.Filesystem's API parity is pretty bad when I was building Mobile Web. About 2 years ago Frankie cleaned up Android's Filesystem so I had hoped that would resolve the discrepancies, however I forgot that iOS's implementation isn't perfect either.

Reply:
I opened up #10128 to address the parity issue and add unit tests here.